### PR TITLE
`Text exercises`: Remove affected submissions feedback warning and automatic feedback tracing

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

With Athena not currently working and the upcoming integrating with the Athena-New repository we want to reduce coupling between Artemis and Athena. This includes features that depend on `text_block` and `text_cluster` information like: The number of affected submissions warning when assessing text exercises and the automatic feedback tracing features. 

They might provide some interesting insights but ultimately, the tutor should grade submission similarly without necessarily paying extra attention on a text block feedback that *might* affect automatic feedback in the future. As discussed with @krusche, we are therefore removing those two features.

### Description

Changes affect the text exercise assessment mode:
- Remove number of affected submissions warning in the textblock feedback editor
- Remove automatic feedback tracing modal and the button that opens it in the textblock feedback editor

❗️Potential impacts on the database:
- Remove `TextAssessmentEventType` enum case: `HOVER_OVER_IMPACT_WARNING` and `VIEW_AUTOMATIC_SUGGESTION_ORIGIN`
- Remove `TextBlock` **transient** property `numberOfAffectedSubmissions`

=> As far as I understand this PR does **not require a migration** and is **safe to use on test servers**. The old assessment events with that type just continue to exist.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Text exercise

Ideally we would test it with Athena running and working which it currently does not (http://artemis-staging.artemis.in.tum.de) -> Use any test server and play around with dev tools

1. Log in to Artemis
2. Participate in a text exercise or find a participation to assess somewhere
3. Go to Course Management
4. Go to the course 
5. Go to the text exercise detail
6. Press "Exercise Dashboard" 
7. If not already done press "Start participating in this exercise" to see all submissions to be assessed
8. Press "Start Assessment" or "Open Assessment" on a submission
9. Check if everything looks good when giving feedback
10. Open dev tools `webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts`
11. Set a breakpoint at `line 88`
12. Add feedback or navigate back and forth to trigger breakpoint (feedback has to exist in that case)
13. *Number of affected submissions warning* -> Dev tools set `this.feedback.credits = 10` and `this.textblock.numberOfAffectedSubmissions = 100`
14. Warning should be removed (c.f. screenshots)
15. Trigger the breakpoint again
16. *Automatic feedback tracing* -> Dev tools set `this.feedback.type = "AUTOMATIC"`
17. Only the automatic badge should be visible now and the search icon button should be removed (c.f. screenshots)

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

#### Manual Tests
- [x] Test 1
- [ ] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [textblock-feedback-editor.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5051-2/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts.html) | 95.52% | ✅ (removed tests) |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [TextAssessmentService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5051-2/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/TextAssessmentService.html) | 100% | ✅ (removed tests) |

#### Number of affected submissions
**Before:** (when giving manual feedback with `textblock.numberOfAffectedSubmissions !== 0` and `feedback.credits !== 0`)
Hard to reproduce without Athena working -> I set `textblock.numberOfAffectedSubmissions = 100` and `feedback.credits = 10` using the dev tools.
Screenshot 2023-06-25 at 17 11 27

**After:** (removed the warning)
Screenshot 2023-06-25 at 17 26 17

#### Automatic feedback tracing
**Before:** (Screenshot from a thesis, hard to reproduce)
feedback-tracing-number-of-affected-submissions-modal

**After:** (No button with search icon next to the automatic badge, associated modal was removed)
Hard to reproduce without Athena working -> I set `feedback.type = "AUTOMATIC"` using the dev tools.
Screenshot 2023-06-25 at 17 25 36

### CODE REVIEW

**Review by simon-hng — DISMISSED** [review]
Code deletions look good to me

File: src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.scss [hunk 1]
```diff
@@ -40,42 +40,6 @@ label {
...
-        background-color: var(--textblock-feedback-editor-feedback-neutral-background-color);
-        color: var(--textblock-feedback-editor-feedback-neutral-color);
-    }
-
-    b {
-        font-weight: 600;
-    }
-}
-
 .white-space {
```
Comment (pal03377) [inline src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.scss:43]: You can also remove the `.white-space` and `.text-inline` styles here.

File: src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts [hunk 2]
```diff
@@ -5,25 +5,10 @@ import { ConfirmIconComponent } from 'app/shared/confirm-icon/confirm-icon.compo
...
-    faExclamation,
-    faExclamationTriangle,
-    faInfoCircle,
-    faQuestionCircle,
-    faRobot,
-    faSearch,
-    faTimes,
-    faTrash,
-} from '@fortawesome/free-solid-svg-icons';
+import { faAngleRight, faBalanceScaleRight, faEdit, faExclamation, faExclamationTriangle, faQuestionCircle, faRobot, faTimes, faTrash } from '@fortawesome/free-solid-svg-icons';
```
Comment (pal03377) [inline src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts:11]: You can also remove the `listOfBlocksWithFeedback` property of the component.

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java [hunk 3]
```diff
@@ -356,8 +356,8 @@ public TextPlagiarismResult createTextPlagiarismResultForExercise(Exercise exerc
     }
 
     public TextAssessmentEvent createSingleTextAssessmentEvent(Long courseId, Long userId, Long exerciseId, Long participationId, Long submissionId) {
-        return TextExerciseFactory.generateTextAssessmentEvent(TextAssessmentEventType.VIEW_AUTOMATIC_SUGGESTION_ORIGIN, FeedbackType.AUTOMATIC, TextBlockType.AUTOMATIC, courseId,
-                userId, exerciseId, participationId, submissionId);
+        return TextExerciseFactory.generateTextAssessmentEvent(TextAssessmentEventType.EDIT_AUTOMATIC_FEEDBACK, FeedbackType.AUTOMATIC, TextBlockType.AUTOMATIC, courseId, userId,
+                exerciseId, participationId, submissionId);
```
Comment (pal03377) [inline src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:360]: I was a bit confused at first, but this is just to generate any text assessment event for tests -> ok

**Review by pal03377 — CHANGES_REQUESTED** [review]
I found some code that can also be removed. Otherwise, this looks good to me! (code)

File: src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.scss (hunk 1 above)
Comment (FelixTJDietrich) [reply src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.scss]: Good catch! I removed it now

File: src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts (hunk 2 above)
Comment (FelixTJDietrich) [reply src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts:11]: Thanks! I removed it :)

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java (hunk 3 above)
Comment (FelixTJDietrich) [reply src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:360]: Yeah me too, but as far as I can see this is fine

**Review by pal03377 — APPROVED** [review]
code

**Review by simon-hng — APPROVED** [review]
Reapprove after css class removal

**Review by pal03377 — APPROVED** [review]
- Code
- Tested on TS4 in Testing Session
✅

**Review by TheZoker — APPROVED** [review]
Tested within testing session on TS4. LGTM 👍

**Review by terlan98 — APPROVED** [review]
Tested on TS5. Works as expected 👍

**Pull request merged**